### PR TITLE
EXIF data as JSON example

### PR DESCRIPTION
**Pull request by raccodee**
Closes (#1)

Set an example of how a JSON download could happen. Test file generated:

```json
{
  "make": "Canon",
  "model": "Canon EOS 40D",
  "date": "2008:07:31 10:38:11",
  "aperture": 7.1,
  "ISO": 100,
  "exposure": 0.00625
}
```

A few suggestions:

1. Take advantage of iteration:

```javascript

// Get the images from the image boxes file1-5
var file1 = document.getElementById('file1').files[0];
var file2 = document.getElementById('file2').files[0];
var file3 = document.getElementById('file3').files[0];
var file4 = document.getElementById('file4').files[0];
var file5 = document.getElementById('file5').files[0];

// Create a new reader for each image
var reader1 = new FileReader();
var reader2 = new FileReader();
var reader3 = new FileReader();
var reader4 = new FileReader();
var reader5 = new FileReader();

```

In this block of code you could just leave a single input for the files and allow multiselect, then iterate over every image in `files` and use one single reader sequentially (less code, less memory provisioned).

```javascript

output.innerHTML += '<p>Make: ' + EXIF.getTag(this, 'Make') + '</p>';
output.innerHTML += '<p>Model: ' + EXIF.getTag(this, 'Model') + '</p>';
output.innerHTML += '<p>Date: ' + EXIF.getTag(this, 'DateTime') + '</p>';
output.innerHTML += '<p>Aperture: ' + EXIF.getTag(this, 'FNumber') + '</p>';
output.innerHTML += '<p>ISO: ' + EXIF.getTag(this, 'ISOSpeedRatings') + '</p>';
output.innerHTML += '<p>Exposure: ' + EXIF.getTag(this, 'ExposureTime') + '</p>';

```

Here, same as before, you could create a `Map` or just a JSON variable with k/v pairs and do the following:

```javascript

const myKVs = {
	Date: "DateTime",
	ISO: "ISOSpeedsRating"
	...
}

Object.keys(myKVs).forEach((k)=>{
	output.innerHTML = `<p>${k}: ${EXIF.getTag(this, myKVs[key])}</p>`
})

```

2. If you're developing a SPA you may want to go with React, Angular, Vue or Svelte. Or maybe compile Pug templates to allow to reuse html code.

But none of this is important as long as it is a proof of concept, those were just my 2 cents 😅

### CODE REVIEW

**Comment by BlockArchitech** [comment]
Awesome! Addressing some things you said:

1: Yes, will do. Currently it's kind of slow due to the way I'm doing things, which isn't great to be honest. If it errors out, it also spams your console and I don't want to do that, so I'm also going to add error handling. I kind of want to try and do it asynchronous but I don't want to make things *too* complicated.

2: I'm looking at react, so I'm probably going to create a new branch and play around with it and see how it fits in this project. [Primer](https://primer.style/react), the design system I'm using, does support react, so it'll be very easy to implement the current style (as you may of noticed, it's very bland, I'm terrible at web design)

Sidenote: You may of noticed, no SSL. My domain blockarchitech.com is through CloudFlare, and I'm having some issues with DNS records not being set. Hopefully, we'll have SSL soon.

I'll go ahead and review this now.

File: assets/js/main.js [hunk 1]
```diff
@@ -7,78 +7,138 @@
...
-	// Get the image from the input box
-	var file = document.getElementById('file').files[0];
-	// Delete the image
-	file.delete();
+  //============================================================
+  // I'm afraid this does nothing: file.delete() is not a function. Javascript will
+  // have the file in memory until the website is reloaded
+  //============================================================
+  //   var file = document.getElementById("file").files[0];
+  //   file.delete();
```
Comment (BlockArchitech) [inline assets/js/main.js:15]: ah, okay. 

So I used something where this *was* a function, but I didn't upload it. If this is useless I'll get rid of it.

File: assets/js/main.js [hunk 2]
```diff
@@ -7,78 +7,138 @@
...
+    {
+      make: EXIF.getTag($thisImage, "Make"),
+      model: EXIF.getTag($thisImage, "Model"),
+      date: EXIF.getTag($thisImage, "DateTime"),
+      aperture: EXIF.getTag($thisImage, "FNumber"),
+      ISO: EXIF.getTag($thisImage, "ISOSpeedRatings"),
+      exposure: EXIF.getTag($thisImage, "ExposureTime"),
+      gps_latitude: EXIF.getTag($thisImage, "GPSLatitude"),
+      gps_longitude: EXIF.getTag($thisImage, "GPSLongitude"),
+      gps_longitude: EXIF.getTag($thisImage, "GPSLongitude"),
```
Comment (BlockArchitech) [inline assets/js/main.js:91]: Is this extra longitude required?

**Review by BlockArchitech — APPROVED** [review]
Thanks! I'll go ahead and merge this, and I can go over some of the stuff I talked about in the other comment.

File: assets/js/main.js (hunk 2 above)
Comment (raccodee) [reply assets/js/main.js:91]: My bad, it’s just duplicated

**Comment by raccodee** [comment]
@BlockArchitech React may be overkill, if we have 1/2 pages or views we can go with Pug/Typescript + gulp setup

**Pull request merged**